packages/flame/lib/src/geometry/shapes/rectangle.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:math';
+import 'dart:ui';
+
+import 'package:meta/meta.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import '../../game/transform2d.dart';
+import 'shape.dart';
+
+/// An axis-aligned rectangle.
```

Comment:
If the `RectangleComponent` should be backed by this later it needs an angle

Reply:
An axis-aligned rectangle is a much more simple shape (so its logic is easier to write and execute). For a rotated rectangle, a Polygon shape would probably be more appropriate.

Reply:
It'll be a bit confusing when the shape components don't map to the shape classes with the same names, but I get your point.

Reply:
Ok, perhaps we are overcomplicating things here. You can define a generic `ShapeComponent` as
```dart
class ShapeComponent extends PositionComponent {
  ShapeComponent(this.shape, this.paint, ...) : super(...);

  final Shape shape;
  final Paint paint;

  @override
  void render(Canvas canvas) {
    canvas.drawPath(shape.toPath(), paint);
  }
}
```
and the `shape` could be anything (including `Rectangle`), and you'd still be able to move/rotate/squeeze it via the PositionComponent's facilities.

Reply:
Remember that these should be the base for the hitboxes too (or they are currently at least), and I don't think we want an API where you always have to wrap a shape with a hitbox component to get a hitbox, instead of having the arguments directly on the hitbox.

Reply:
And also, I think we should avoid doing a breaking change again here since we have just refactored this API.

Reply:
> Remember that these should be the base for the hitboxes too (or they are currently at least)

A `ShapeComponent` is a lightweight component which draws a geometrical shape using a simple paint. It shouldn't be the base for hitboxes. The only reason a Hitbox is based on a `ShapeComponent` right now is that shapes don't exist as standalone reusable entities, instead they are "baked in" into the `ShapeComponent`. 

Which is exactly what this PR is trying to address: create `Shape` as a separate abstract entity, and then let various components use those shapes as they need.